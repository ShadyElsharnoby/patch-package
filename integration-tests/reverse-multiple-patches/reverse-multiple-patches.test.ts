import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "reverse-multiple-patches",
  shouldProduceSnapshots: true,
})
