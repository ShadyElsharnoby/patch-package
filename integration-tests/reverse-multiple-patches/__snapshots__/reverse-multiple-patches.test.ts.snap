// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test reverse-multiple-patches: 00: the patches were applied 1`] = `
"SNAPSHOT: the patches were applied
'goodbye world';
END SNAPSHOT"
`;

exports[`Test reverse-multiple-patches: 01: --reverse undoes the patches 1`] = `
"SNAPSHOT: --reverse undoes the patches
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (3 goodbye) ✔
left-pad@1.3.0 (2 world) ✔
left-pad@1.3.0 (1 hello) ✔
END SNAPSHOT"
`;

exports[`Test reverse-multiple-patches: 02: The patches can be reapplied 1`] = `
"SNAPSHOT: The patches can be reapplied
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔
left-pad@1.3.0 (2 world) ✔
left-pad@1.3.0 (3 goodbye) ✔
END SNAPSHOT"
`;

exports[`Test reverse-multiple-patches: 03: if one of the patches fails then reverse only undoes the ones that succeeded 1`] = `
"SNAPSHOT: if one of the patches fails then reverse only undoes the ones that succeeded
patches/left-pad+1.3.0+003+goodbye.patch
 9: -'use schmorld';
 10: +'goodbye schmorld';
apply broken
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔
left-pad@1.3.0 (2 world) ✔

⛔ ERROR

Failed to apply patch file left-pad+1.3.0+003+goodbye.patch.

If this patch file is no longer useful, delete it and run

  patch-package

To partially apply the patch (if possible) and output a log of errors to fix, run

  patch-package --partial

After which you should make any required changes inside node_modules/left-pad, and finally run

  patch-package left-pad

to update the patch file.

reverse all but broken
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (2 world) ✔
left-pad@1.3.0 (1 hello) ✔
END SNAPSHOT"
`;
